libs/mrwebrtc-unityplugin/src/native_renderer.cpp

```diff
@@ -0,0 +1,271 @@
+  width = frame.width_;
+  height = frame.height_;
+  ystride = frame.ystride_;
+  ustride = frame.ustride_;
+  vstride = frame.vstride_;
+  size_t ysize = (size_t)ystride * height;
+  size_t usize = (size_t)ustride * height / 2;
+  size_t vsize = (size_t)vstride * height / 2;
+  ybuffer.resize(ysize);
+  ubuffer.resize(usize);
+  vbuffer.resize(vsize);
+  memcpy(ybuffer.data(), frame.ydata_, ysize);
+  memcpy(ubuffer.data(), frame.udata_, usize);
+  memcpy(vbuffer.data(), frame.vdata_, vsize);
+}
+
+NativeRenderer* NativeRenderer::Create(mrsRemoteVideoTrackHandle videoTrackHandle) {
+  {
+    // Global lock
+    std::lock_guard guard(g_lock);
+    auto nativeVideo = new NativeRenderer(videoTrackHandle);
+    g_nativeVideos.insert(nativeVideo);
+    return nativeVideo;
+  }
+}
+
+void NativeRenderer::Destroy(mrsNativeVideoHandle nativeVideoHandle) {
+  {
+    // Global lock
+    std::lock_guard guard(g_lock);
+    NativeRenderer* nativeVideo = (NativeRenderer*)nativeVideoHandle;
+    nativeVideo->Shutdown();
+    g_nativeVideos.erase(nativeVideoHandle);
+  }
+}
+
+NativeRenderer::NativeRenderer(mrsRemoteVideoTrackHandle videoTrackHandle) : m_handle(videoTrackHandle) {
+  Log_Debug("NativeRenderer::NativeRenderer");
+  if (!g_renderApi) {
+    Log_Warning("NativeRenderer: Unity plugin not initialized.");
+  }
+}
+
+NativeRenderer::~NativeRenderer() {
+  /// This is unsafe to log in Unity Editor.
+  // Log_Debug("NativeRenderer::~NativeRenderer");
+}
+
+void NativeRenderer::Shutdown() {
+  Log_Debug("NativeRenderer::Shutdown");
+  DisableRemoteVideo();
+}
+
+void NativeRenderer::EnableRemoteVideo(VideoKind format,
+                                        TextureDesc textureDescs[],
+                                        int textureDescCount) {
+  if (!g_renderApi) {
+    Log_Warning("NativeRenderer: Unity plugin not initialized.");
+  }
+  {
+    // Instance lock
+    std::lock_guard guard(m_lock);
+    m_remoteVideoFormat = format;
+    switch (format) {
+      case VideoKind::kI420:
+        if (textureDescCount == 3) {
+          m_remoteTextures.resize(3);
+          m_remoteTextures[0] = textureDescs[0];
+          m_remoteTextures[1] = textureDescs[1];
+          m_remoteTextures[2] = textureDescs[2];
+          mrsRemoteVideoTrackRegisterI420AFrameCallback(
+              m_handle, 
+              NativeRenderer::I420ARemoteVideoFrameCallback,
+              this);
+        }
+        break;
+
+      case VideoKind::kARGB:
+        Log_Warning("NativeRenderer: kARGB not currently supported.");
+        // TODO
+        break;
+
+      case VideoKind::kNone:
+        Log_Warning("NativeRenderer: No VideoKind specified.");
+        // TODO
+        break;
+    }
+  }
+}
+
+void NativeRenderer::DisableRemoteVideo() {
+  Log_Debug("NativeRenderer::DisableRemoteVideo");
+  {
+    // Instance lock
+    std::lock_guard guard(m_lock);
+    m_remoteTextures.clear();
+    m_remoteVideoFormat = VideoKind::kNone;
+  }
```

Comment:
This should call `mrsRemoteVideoTrackRegisterI420AFrameCallback(m_handle, nullptr, nullptr)`.